src/hazelweb/gui/ActionPanel.re

```diff
@@ -0,0 +1,464 @@
+    action_label(
+      ~attrs=[
+        Vdom.Attr.(
+          style(
+            Css_gen.(
+              create(~field="display", ~value="flex")
+              @> create(~field="justify-content", ~value="space-between")
+            ),
+          )
+        ),
+      ],
+      children,
+    );
+  };
+
+  let dual_line = (description1, combo1, description2, combo2) => {
+    spaced_line([
+      Vdom.Node.div(
+        [display_inline_block],
+        [text(description1), keyboard_button(combo1)],
+      ),
+      Vdom.Node.div(
+        [display_inline_block],
+        [text(description2), keyboard_button(combo2)],
+      ),
+    ]);
+  };
+
+  let bottom_border_none =
+    Css_gen.create(~field="border-bottom", ~value="none");
+
+  let add_style = (style, elem) => {
+    switch (elem) {
+    | Vdom.Node.(Element(e)) =>
+      Vdom.Node.(Element(Element.add_style(e, style)))
+    | n => n
+    };
+  };
+
+  KeyCombo.[
+    section(
+      "Movement",
+      [
+        info([text("Move using arrow keys")]),
+        dual_line("Move to next hole ", Tab, "Previous hole ", ShiftTab),
+      ],
+    ),
+    section(
+      "General Editing",
+      [
+        spaced_line([
+          text("Backspace / Delete"),
+          Vdom.Node.div(
+            [display_inline_block],
+            [
+              keyboard_button(Backspace) |> add_style(bottom_border_none),
+              keyboard_button(Delete) |> add_style(bottom_border_none),
+            ],
+          ),
+        ]),
+        combo(Ctrl_Alt_I, simple("Swap expression up")),
+        combo(Ctrl_Alt_K, simple("Swap expression down")),
+        combo(Ctrl_Alt_J, simple("Swap expression left")),
+        combo(Ctrl_Alt_L, simple("Swap expression right")),
+        combo(Enter, simple("Create new line ")),
+        combo(LeftParen, simple("Parenthesize")),
+      ],
+    ),
+    section(
+      "Arithmetic",
+      [
+        info([
+          text("Enter number literals directly e.g. "),
+          mono_text("1.0, 2"),
+        ]),
+        info_action(
+          [
+            text("Type "),
+            mono_text("\"I\""),
+            text(" to insert an Integer type"),
+          ],
+          Action.Construct(SChar("I")),
+        ),
+        info_action(
+          [
+            text("Type "),
+            mono_text("\"F\""),
+            text(" to insert a Float type"),
+          ],
+          Action.Construct(SChar("F")),
+        ),
+        spaced_line([
+          text("Floating point operators, type"),
+          Vdom.Node.div(
+            [],
+            [
+              brown_label([mono_text("+.")]) |> add_style(bottom_border_none),
+              brown_label([mono_text("-.")])
+              |> add_style(bottom_border_none),
+              brown_label([mono_text("*.")])
+              |> add_style(bottom_border_none),
+            ],
+          ),
+        ]),
+        operator_list(
+          ~on_type=false,
+          "Integer operators",
```

Comment:
Put integer operators first, to be consistent with order of the type insertion lines